clang/lib/AST/Mangle.cpp

```diff
@@ -74,7 +74,7 @@ static CCMangling getCallingConvMangling(const ASTContext &Context,
       if (FD->isMain() && FD->getNumParams() == 2)
         return CCM_WasmMainArgcArgv;
 
-  if (!Triple.isOSWindows() || !Triple.isX86())
+  if (!Triple.isOSWindowsOrUEFI() || !Triple.isX86())
```

Comment:
As we discussed offline, I'd really rather see all sites like this use a predicate that's specific to what matters to the call site in precise terms.  This predicate is really an answer to the question rather than the question that should be asked.  It presumes the answer is "Windows or UEFI targets", when the question it should be asking is something more like, "What C++ ABI name mangling regime is in use by this target?"

Reply:
To make things more complicated, consider mingw. I am totally unfamiliar to UEFI, but it seems like folks want it to inherit lots of MSVC ABI features: the mangling scheme, the vftable ABI rules, the struct layout, etc. These rules are usually conditional on `isWindowsMSVCEnvironment()`, so you'd need to audit those conditions.

While explicit triple checks are convenient and necessary in many cases, we generally try to abstract these details with the various CXXABI interfaces, and if you can find a way to do that, it may help with readability.